### ISSUE TITLE
bulk i/o: accelerate testing of backup/restore and import/export

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#52464]
Our backup/restore and export/import tests make heavy use of "table fingerprinting" (via `SHOW EXPERIMENTAL_FINGERPRINTS`): check that the data after a restore is the same that was backed up.

This is used both in unit tests and in roachtests.

Today, the fingerprinting uses the FNV hashing function for table data. This function was chosen because it is much faster than cryptographically secure hashes like SHA, yet has better collision properties than e.g. a simple XOR.

There has been recent development on super-fast hashing since FNV though. For example, the state-of-the art is currently xxHash (https://cyan4973.github.io/xxHash/) which operates [between 2 and  30 (!) times faster than FNV](https://github.com/Cyan4973/xxHash#benchmarks) on conventional hardware.

By upgrading SHOW EXPERIMENTAL_FINGERPRINTS to use e.g. xxHash instead of FNV, we would be able to accelerate the consistency check in the bulk i/o tests accordingly.

cc @mwang1026 @dt for triage. I think the effort is extremely modest with a potential high upside (productivity for the team by accelerating testing).

Epic CRDB-7779

Jira issue: CRDB-3947

**Comment by dt** [comment]
I would be (pleasantly) quite surprised if the cpu cost of the actual hashing dominates the overall cost of fingerprinting — when last we looked at this, the cost was very much in our code, retrieving the data, versus actually hashing it.

Another concern would be compatibility: if the point of a fingerprint is to tell if it changed, Channing the fingerprint could be bad, but I think we gave ourselves room to do that with EXPERIMENTAL.

FWIW I think we could also maybe just remove that function entirely now that our SQL layer has become smarter and supports variadic aggregates, and run something like `SELECT hash_fnv(*) FROM my_tbl` instead. Then the caller can choose their desired function too.
